yarn 4 installation fails

While yarn classic installation works, yarn 4 gives me this:
```
➜  Desktop yarn create rescript-app
➤ YN0000: · Yarn 4.1.1
➤ YN0000: ┌ Resolution step
➤ YN0085: │ + create-rescript-app@npm:1.8.0
➤ YN0000: └ Completed
➤ YN0000: ┌ Fetch step
➤ YN0013: │ A package was added to the project (+ 347.13 KiB).
➤ YN0000: └ Completed
➤ YN0000: ┌ Link step
➤ YN0000: └ Completed
➤ YN0000: · Done in 0s 44ms

node:internal/process/esm_loader:34
      internalBinding('errors').triggerUncaughtException(
                                ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/private/var/folders/4z/fmty9jsx2nz0b77clgyfjj_r0000gn/T/xfs-db3e3751/dlx-63482/.yarn/cache/create-rescript-app-npm-1.8.0-4abaf919f9-1cd44032c4.zip/node_modules/create-rescript-app/out/create-rescript-app.mjs' imported from /Users/florian-cca/Desktop/
Did you mean to import /private/var/folders/4z/fmty9jsx2nz0b77clgyfjj_r0000gn/T/xfs-db3e3751/dlx-63482/.yarn/cache/create-rescript-app-npm-1.8.0-4abaf919f9-1cd44032c4.zip/node_modules/create-rescript-app/out/create-rescript-app.mjs?
    at finalizeResolution (node:internal/modules/esm/resolve:264:11)
    at moduleResolve (node:internal/modules/esm/resolve:917:10)
    at defaultResolve (node:internal/modules/esm/resolve:1130:11)
    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:396:12)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:365:25)
    at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:240:38)
    at ModuleLoader.import (node:internal/modules/esm/loader:328:34)
    at node:internal/modules/run_main:99:28
    at loadESM (node:internal/process/esm_loader:28:13)
    at runMainESM (node:internal/modules/run_main:98:21) {
  code: 'ERR_MODULE_NOT_FOUND',
  url: 'file:///private/var/folders/4z/fmty9jsx2nz0b77clgyfjj_r0000gn/T/xfs-db3e3751/dlx-63482/.yarn/cache/create-rescript-app-npm-1.8.0-4abaf919f9-1cd44032c4.zip/node_modules/create-rescript-app/out/create-rescript-app.mjs'
}

Node.js v20.11.1
```

--- a/.gitignore
+++ b/.gitignore
@@ -3,4 +3,4 @@
 **/*.res.js
 **/*.res.mjs
 .bsb.lock
-out/create-rescript-app.mjs
+out/create-rescript-app.cjs

--- a/package.json
+++ b/package.json
@@ -2,7 +2,7 @@
   "name": "create-rescript-app",
   "version": "1.10.0-beta.1",
   "description": "Quickly create new ReScript apps from project templates.",
-  "main": "out/create-rescript-app.mjs",
+  "main": "out/create-rescript-app.cjs",
   "scripts": {
     "start": "node src/Main.res.mjs",
     "prepack": "rescript && rollup -c",
@@ -30,10 +30,10 @@
   ],
   "files": [
     "templates",
-    "out/create-rescript-app.mjs"
+    "out/create-rescript-app.cjs"
   ],
   "bin": {
-    "create-rescript-app": "out/create-rescript-app.mjs"
+    "create-rescript-app": "out/create-rescript-app.cjs"
   },
   "devDependencies": {
     "@clack/prompts": "^0.11.0",

--- a/rollup.config.mjs
+++ b/rollup.config.mjs
@@ -5,8 +5,8 @@ import terser from "@rollup/plugin-terser";
 export default {
   input: `src/Main.res.mjs`,
   output: {
-    file: `out/create-rescript-app.mjs`,
-    format: "es",
+    file: `out/create-rescript-app.cjs`,
+    format: "cjs",
     banner: "#!/usr/bin/env node",
   },
   plugins: [terser(), nodeResolve({ preferBuiltins: true }), commonjs()],

--- a/src/Main.res
+++ b/src/Main.res
@@ -18,7 +18,7 @@ let handleError = async (~outro, perform) =>
     Process.exitWithCode(1)
   }
 
-let run = async () => {
+let main = async () => {
   let version = await getVersion()
   P.intro(C.dim(`create-rescript-app ${version}`))
 
@@ -66,6 +66,10 @@ https://rescript-lang.org`,
   }
 }
 
-try await run() catch {
-| P.Canceled => P.cancel("Canceled.")
-}
+// Do not use top-level await, otherwise we can't package as .cjs.
+let run = async () =>
+  try await main() catch {
+  | P.Canceled => P.cancel("Canceled.")
+  }
+
+run()->Promise.ignore

